### ISSUE TITLE
NameError: referenced before assignment in enclosing scope

### ISSUE THREAD

**Issue by xoriole** [Tribler/tribler#3829]
From reporter - Tribler 7.1.0-beta (Linux-4.18.5-arch1-1-ARCH-x86_64-with-glibc2.2.5) [#15828](http://reporter.tribler.org/stacktrace/15828)
```python
Traceback (most recent call last):
File "/usr/share/tribler/TriblerGUI/tribler_request_manager.py", line 388, in <lambda>
network_reply.finished.connect(lambda: reply_callback(network_reply, log))
NameError: free variable 'reply_callback' referenced before assignment in enclosing scope
```

**Comment by qstokkink** [comment]
I think this should be solved by first assigning to a local variable, something like:

```python
the_reply_callback = reply_callback
 ...
network_reply.finished.connect(lambda: the_reply_callback (network_reply, log))
```

**Comment by qstokkink** [comment]
I cannot reproduce this though: when running Tribler I only ever see `reply_callback` in `locals()` and not in `globals()`.

**Comment by qstokkink** [comment]
Aha, this is probably due to someone running with Python `3.x`.

**Comment by devos50** [comment]
[This one is crashing the Tribler application tester](https://jenkins-ci.tribler.org/job/_Nightly_Build_6_Hours/7/). I will move it to the 7.2 milestone.

**Comment by devos50** [comment]
Some progress with the fix in #4086, getting a different error now:

```
test_add_download_url (Tribler.Test.GUI.test_gui.TriblerGUITest) ... Traceback (most recent call last):
  File "C:\workspace\jenkins2\workspace\GH_Tribler_PR_Tests\PR_GUI_win64\tribler\TriblerGUI\tribler_request_manager.py", line 388, in <lambda>
    network_reply.finished.connect(lambda cb=reply_callback: cb(network_reply, log))
NameError: free variable 'network_reply' referenced before assignment in enclosing scope
```

**Comment by devos50** [comment]
Most likely fixed by #4086 since I haven't observed this error in the nightly deployment tests.
